pyani_plus/db_orm.py

```diff
@@ -654,11 +655,20 @@ def db_genome(
 
     length = 0
     description = None
-    with Path(fasta_filename).open() as handle:
-        for title, seq in SimpleFastaParser(handle):
-            length += len(seq)
-            if description is None:
-                description = title  # Just use first entry
+
+    try:
+        with gzip.open(fasta_filename, "rt") as handle:
+            for title, seq in SimpleFastaParser(handle):
+                length += len(seq)
+                if description is None:
+                    description = title  # Just use first entry
+    except gzip.BadGzipFile:
+        with Path(fasta_filename).open() as handle:
+            for title, seq in SimpleFastaParser(handle):
+                length += len(seq)
+                if description is None:
+                    description = title  # Just use first entry
+
```

Comment:
This try/except style for detecting gzip is used often enough that a helper function might be worthwhile, ideally usable as a context manager?